python/interpret_community/explanation/serialization.py

```diff
@@ -0,0 +1,306 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""Defines the save and load explanation methods and helpers for serialization."""
+
+import json
+import os
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix, issparse
+
+from ..common.constants import ExplainParams, ExplainType
+from ..dataset.dataset_wrapper import DatasetWrapper
+from ..explanation import explanation as patched_explanation
+from .explanation import (ClassesMixin, FeatureImportanceExplanation,
+                          GlobalExplanation, LocalExplanation,
+                          _create_global_explanation,
+                          _create_local_explanation)
+
+EXPLANATION_METADATA = 'explanation_metadata.json'
+SERIALIZATION_VERSION = 'serialization_version'
+
+
+def save_explanation(explanation, path, exist_ok=False):
+    """Serialize the explanation.
+
+    :param explanation: The Explanation to be serialized.
+    :type explanation: Explanation
+    :param path: The path to the directory in which the explanation will be saved. By default, must be a new directory
+        to avoid overwriting any previous explanations. Set exist_ok to True to overrule this behavior.
+    :type path: str
+    :param exist_ok: If False (default), the path provided by the user must not already exist and will be created by
+        this function. If True, a prexisting path may be passed. Any preexisting files whose names match those of the
+        files that make up the explanation will be overwritten.
+    :type exist_ok: bool
+    """
+    if os.path.exists(path) and not exist_ok:
+        raise Exception('The directory specified by path already exists. '
+                        'Please pass in a new directory or set exists_ok=True.')
+    os.makedirs(path, exist_ok=True)
+
+    # TODO replace with set of params from below
```

Comment:
TODO for this PR or another?

Reply:
another PR, this is just a comment from previous code